src/WorkBC.Web/Controllers/CareerProfilesController.cs

```diff
@@ -52,20 +52,25 @@ public async Task<ActionResult<IEnumerable<CareerProfileModel>>> GetSavedCareerP
                     s => s.First().Id
                 );
 
-            List<int> savedCareerProfilesIds = savedCareerProfilesDict.Select(s => s.Key).ToList();
+            var savedCareerProfilesIds = savedCareerProfilesDict.Select(s => s.Key).ToList();
+            var careerProfiles = await _ssotApi.GetCareerProfiles(savedCareerProfilesIds);
+            
+            var results = new List<CareerProfileModel>();
 
-            IQueryable<CareerProfileModel> query = from p in _enterpriseContext.CareerProfiles
-                join n in _enterpriseContext.Nocs on p.NocId equals n.NocId
-                orderby n.NameEnglish
-                where savedCareerProfilesIds.Contains(p.CareerProfileId)
-                select new CareerProfileModel
-                {
-                    Id = savedCareerProfilesDict[p.CareerProfileId],
-                    Title = n.NameEnglish,
-                    NocCode = n.Noccode
-                };
-
-            return Ok(await query.ToListAsync());
+            if (careerProfiles.Count > 0)
```

Comment:
Don't need this if condition because the loop will not run if the list is empty.